### ISSUE TITLE
"CrossEntropyLoss" should mention in its name that it takes softmax for target

### ISSUE THREAD

**Issue by IJCAI2019-985** [pytorch/pytorch#22604]
https://stackoverflow.com/questions/49390842/cross-entropy-in-pytorch
https://discuss.pytorch.org/t/why-does-crossentropyloss-include-the-softmax-function/4420
https://discuss.pytorch.org/t/do-i-need-to-use-softmax-before-nn-crossentropyloss/16739

people keep asking questions like this for the name is not clear

**Comment by soumith** [comment]
this is a fair ask, we should probably add SoftmaxCrossEntropyLoss, but we shouldn't deprecate CrossEntropyLoss due to historical importance
